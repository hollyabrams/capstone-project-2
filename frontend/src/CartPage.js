import React, { useState } from 'react';
import { useShoppingCart } from 'use-shopping-cart';
import { Link } from 'react-router-dom';
import CartProduct from './components/CartProduct';
import CheckoutForm from './forms/CheckoutForm';
import Modal from 'react-modal';

// Bind modal to your appElement (http://reactcommunity.org/react-modal/accessibility/)
Modal.setAppElement('#root')

function CartPage() {
    const { cartCount, cartDetails, formattedTotalPrice, totalPrice, clearCart } = useShoppingCart();
    const [showCheckout, setShowCheckout] = useState(false);
    const currency = Object.values(cartDetails)[0]?.currency || 'usd';

    return (
        <div className="container xl:max-w-screen-xl mx-auto py-12 px-20">
            {cartCount > 0 ? (
                <>
                    <h2 className="text-4xl font-semibold">Your shopping cart</h2>
                    <p className="mt-1 text-xl">
                        {cartCount} items{" "}
                        <button className="opacity-50 hover:opacity-100 text-base capitalize " onClick={() => clearCart()}>(Clear all)</button>
                    </p>
                </>
            ) : (
                <>
                    <h2 className="text-4xl font-semibold">
                        Your shopping cart is empty.
                    </h2>
                    <p className="mt-1 text-xl">
                        Check out our awesome products{" "}
                        <Link to="/products" className="text-red-500 underline">here!</Link>
                    </p>
                </>
            )}

            {cartCount > 0 && <div className="mt-12 space-y-4">
                {Object.entries(cartDetails).map(([productId, product]) => (
                    <CartProduct key={productId} product={product} />
                ))}

                <div className="flex flex-col items-end border-t py-4 mt-8">
                    <p className="text-xl">Total:{" "}
                    <span className="font-semibold">{formattedTotalPrice}</span>
                    </p>
                    <button 
                    onClick={() => setShowCheckout(true)} // set showCheckout to true when clicked
                    className="border rounded py-2 px-6 bg-yellow-500 hover:bg-yellow-600 border-yellow-500 hover:border-yellow-600 focus:ring-4 focus:ring-opacity-50 focus:ring-yellow-500 text-white transition-colors disabled:opacity-50 disabled:cursor-not-allowed disabled:hover:bg-yellow-500 mt-4 max-w-max">
                    Checkout
                </button>
                </div>
                <Modal
                    isOpen={showCheckout}
                    onRequestClose={() => setShowCheckout(false)}
                    style={{
                        overlay: { backgroundColor: 'rgba(0, 0, 0, 0.75)' },
                        content: {
                        color: 'lightsteelblue',
                        width: '50%',
                        height: '50%', 
                        marginLeft: 'auto',
                        marginRight: 'auto',
                        marginTop: 'auto',
                        marginBottom: 'auto'
                        }
                    }}
                    contentLabel="Checkout Modal"
                    >
                    <CheckoutForm amount={totalPrice} currency={currency}/>
                </Modal>
            </div>}
        </div>
    )
}

export default CartPage;

