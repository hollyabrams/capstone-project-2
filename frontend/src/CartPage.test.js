import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import CartPage from './CartPage';

// Mock the useShoppingCart hook
jest.mock('use-shopping-cart', () => ({
  useShoppingCart: () => ({
    cartCount: 0,
    cartDetails: {},
    formattedTotalPrice: "$0.00",
    totalPrice: 0,
    clearCart: jest.fn(),
  }),
}));

// Mock the Modal setAppElement function
jest.mock('react-modal', () => {
  const originalModule = jest.requireActual('react-modal');
  originalModule.setAppElement = jest.fn(); // do nothing function
  return originalModule;
});

describe('CartPage', () => {
  test('renders without errors', () => {
    render(
      <Router>
        <CartPage />
      </Router>
    );
  });

  test('matches snapshot', () => {
    const { asFragment } = render(
      <Router>
        <CartPage />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
