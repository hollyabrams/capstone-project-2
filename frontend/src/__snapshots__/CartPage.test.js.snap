// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartPage matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container xl:max-w-screen-xl mx-auto py-12 px-20"
  >
    <h2
      class="text-4xl font-semibold"
    >
      Your shopping cart is empty.
    </h2>
    <p
      class="mt-1 text-xl"
    >
      Check out our awesome products 
      <a
        class="text-red-500 underline"
        href="/products"
      >
        here!
      </a>
    </p>
  </div>
</DocumentFragment>
`;
